.github/workflows/_unit_test.yml

```diff
@@ -60,18 +61,47 @@ jobs:
     - id: set-batches
       name: Distribute tasks into batches
       run: |
-        batches_json=$(bundle exec rake github:generate_batches)
-        echo "$batches_json" | ruby -rjson -e 'puts JSON.pretty_generate(JSON.parse(STDIN.read))'
-        echo "batches=$batches_json" >> $GITHUB_OUTPUT
+        data=$(bundle exec rake github:generate_batches)
+        echo "$data" | ruby -rjson -e 'puts JSON.pretty_generate(JSON.parse(STDIN.read))'
+
+        # Extract each key and set it as a separate output
+        batches_data=$(echo "$data" | ruby -rjson -e 'puts JSON.parse(STDIN.read)["batches"].to_json')
+        misc_data=$(echo "$data" | ruby -rjson -e 'puts JSON.parse(STDIN.read)["misc"].to_json')
+
+        echo "batches=$batches_data" >> $GITHUB_OUTPUT
+        echo "misc=$misc_data" >> $GITHUB_OUTPUT
     - name: Generate batch summary
       run: bundle exec rake github:generate_batch_summary
       env:
         batches_json: "${{ steps.set-batches.outputs.batches }}"
-  build-test:
+
+  # `Initialize containers` step becomes quite heavily when many services are used.
+  #
+  # The job can failed with timeout because it takes a long time pulling the image or waiting for the service to be ready).
+  #
+  # `build-test-standard` job is used to run most of our tasks and configured with the following services:
+  #   - mysql
+  #   - postgres
+  #   - redis
+  #
+  # `build-test-misc` job is extracted to run specific tasks that requires the following services:
+  #   - elasticsearch
+  #   - memcached
+  #   - mongodb
+  #   - opensearch
+  #   - presto
+  #
+  #  Benefit of this optimization:
+  #  - Unnecessary services are not started
+  #  - Reduce the overhead for pulling images
+  #  - Improve reliability for starting dependent services
+  #
+  # In the future, we could consider to extract jobs to run database dependent tasks to optimize the workflow.
```

Comment:
I was going to suggest this is a good first step but I'd like `build-test-misc` to be split into five so that each one only pulls their relevant service.

Seems like improvements are on the table as next steps, so let's roll with this PR.